### PR TITLE
fix(lyrics): correct TTML background-vocal cue timing and whitespace

### LINKED ISSUE #5671: [Bug]: TTML parsing with new songLyrics v2 for background agent is a bit weird
### I confirm that:

- [x] I have searched the existing [open AND closed issues](https://github.com/navidrome/navidrome/issues?q=is%3Aissue) to see if an issue already exists for the bug I've encountered
- [x] I'm using the latest version (your issue may have been fixed already)

### Version

0.62.0-SNAPSHOT (bd9fa1c602ee2c20fce23abadd6236cdc35e71b5)

### Current Behavior

The ttml lyrics parsed into this:

<details>
<summary>Snippet of parsed TTML background lines</summary>

```json
{
    "index": 22,
    "start": 82889,
    "end": 86859,
    "value": "It go in, out, do it real slow  (When you ride it, slide)",
    "agentId": "v2",
    "cue": [
    {
        "start": 82889,
        "end": 83127,
        "byteStart": 0,
        "byteEnd": 1,
        "value": "It"
    },
    {
        "start": 83127,
        "end": 83374,
        "byteStart": 3,
        "byteEnd": 4,
        "value": "go"
    },
    {
        "start": 83374,
        "end": 83938,
        "byteStart": 6,
        "byteEnd": 8,
        "value": "in,"
    },
    {
        "start": 84210,
        "end": 85066,
        "byteStart": 10,
        "byteEnd": 13,
        "value": "out,"
    },
    {
        "start": 85066,
        "end": 85593,
        "byteStart": 15,
        "byteEnd": 19,
        "value": "do it"
    },
    {
        "start": 85593,
        "end": 85934,
        "byteStart": 21,
        "byteEnd": 24,
        "value": "real"
    },
    {
        "start": 85934,
        "end": 85934,
        "byteStart": 26,
        "byteEnd": 29,
        "value": "slow"
    }
    ]
},
{
    "index": 22,
    "start": 82889,
    "end": 86859,
    "value": "It go in, out, do it real slow  (When you ride it, slide)",
    "agentId": "__nd_bg__|v2",
    "cue": [
    {
        "start": 83881,
        "end": 84243,
        "byteStart": 32,
        "byteEnd": 40,
        "value": "(When you"
    },
    {
        "start": 84669,
        "end": 85123,
        "byteStart": 42,
        "byteEnd": 49,
        "value": "ride it,"
    },
    {
        "start": 86232,
        "end": 86859,
        "byteStart": 51,
        "byteEnd": 56,
        "value": "slide)"
    }
    ]
},
```

</details>

The issue:
- `value` for the main agent include stuff from `x-bg` role (same with bg agent where it include the main agent content)
- The last cue line for the main agent has the same `start` and `end` value

### Expected Behavior

The `value` for background agent is not the full line, but just the background part

This made the Symfonium lyrics viewer duplicates the line like this (Purple should be bg agent and orange-ish should be main)

<details>
<summary>Symfonium screenshot</summary>

Image

</details>

The original TTML are this:
```xml
<p begin="1:22.889" end="1:26.859" itunes:key="L23" ttm:agent="v2">
    <span begin="1:22.889" end="1:23.127">It</span>
    <span begin="1:23.127" end="1:23.374">go</span>
    <span begin="1:23.374" end="1:23.938">in,</span>
    <span begin="1:24.210" end="1:25.066">out,</span>
    <span begin="1:25.066" end="1:25.593">do it</span>
    <span begin="1:25.593" end="1:25.934">real</span>
    <span begin="1:25.934" end="1:26.751">slow</span>
    <span ttm:role="x-bg">
        <span begin="1:23.881" end="1:24.243">(When you</span>
        <span begin="1:24.669" end="1:25.123">ride it,</span>
        <span begin="1:26.232" end="1:26.859">slide)</span>
    </span>
</p>
```

### Steps To Reproduce

1. Add the following file [03. luther.xml](https://github.com/user-attachments/files/29399358/03.luther.xml) (Rename the extension to `ttml` since GitHub doesn't support it apparently) into your Navidrome instance
2. Request the getSongLyricsById OS endpoint with enhanced set to true

### Environment

```markdown
- OS: Ubuntu 24.04.4 LTS
- Browser: Firefox/Zen 151.0.4
- Client: Symfonium 15.0.0 Beta 6
```

### How Navidrome is installed?

Binary (from downloads page)

### Configuration

```toml
DataFolder = "REDACTED"
MusicFolder = "REDACTED"

Port = 45263
Address = "127.0.0.1"

LastFM.ApiKey = "REDACTED"
LastFM.Secret = "REDACTED"

EnableTranscodingConfig = false
TranscodingCacheSize = "1GiB"

EnableInsightsCollector = false

Agents = "audiomuseai,apple-music,deezer,lastfm"

Scanner.ScanOnStartup = false
Scanner.PurgeMissing = "always"

# PID
PID.Album = "musicbrainz_albumid|albumartistid,album,albumversion"

# Tagging
Tags.ISRC.Aliases = ["isrc", "----:com.apple.itunes:isrc"]
Tags.ISRC.Album = true
Tags.Copyright.Aliases = ["copyright"]
Tags.Copyright.Album = true
Tags.Comment.Aliases = ["comment"]
Tags.Comment.Album = false
Tags.Artist.Split = [" / ", " feat. ", " feat ", " ft. ", " ft ", "; ", " & ", " × ", "、 "]
#Tags.Artists.Aliases = ["txxx:artists", "artists", "----:com.apple.itunes:artists", "wm/artists", "©art"]
# Tags.Artists.Split = ["/", " / ", " feat. ", " feat ", " ft. ", " ft ", "; ", " & ", "×", " × "]
Tags.AlbumArtist.Split = [" / ", " feat. ", " feat ", " ft. ", " ft ", "; ", " & ", " × ", "、 "]
#Tags.AlbumArtists.Aliases = ["txxx:album artists", "albumartists", "aart"]
Tags.AppleAlbum.Aliases = ["plid", "----:com.apple.itunes:itunesalbumid", "itunesalbumid"]
Tags.AppleAlbum.Album = true
Tags.AppleTrack.Aliases = ["cnid"]
Tags.AppleTrack.Album = false
```

### Relevant log output

```shell
Jun 26 23:49:37 helios navidrome[3427901]: time="2026-06-26T23:49:37Z" level=info msg="Watcher: Triggering scan for changed folders" numTargets=1
Jun 26 23:49:37 helios navidrome[3429629]: time="2026-06-26T23:49:37Z" level=info msg="Loaded configuration" file=/etc/navidrome/navidrome.toml
Jun 26 23:49:37 helios navidrome[3429629]: time="2026-06-26T23:49:37Z" level=info msg="Scanning specific folders" numTargets=1
Jun 26 23:49:37 helios navidrome[3429629]: time="2026-06-26T23:49:37Z" level=info msg="Scanner: Starting selective scan" fullScan=false numLibraries=1 numTargets=1
Jun 26 23:49:37 helios navidrome[3429629]: time="2026-06-26T23:49:37Z" level=info msg="Scanner: Completed processing folder" audioCount=12 elapsed=3.8ms imageCount=1 library=Main plsCount=0 tracksImported=0 tracksMissing=0 ​folder="English/Kendrick Lamar/[2024.11.22] GNX [Album ALAC 48 24]"
Jun 26 23:49:37 helios navidrome[3429629]: time="2026-06-26T23:49:37Z" level=info msg="Scanner: Finished scanning selected folders" duration=161.6ms numTargets=1
Jun 26 23:49:37 helios navidrome[3427901]: time="2026-06-26T23:49:37Z" level=info msg="Watcher: Scan completed
```

### Anything else?

Here's the full parsed TTML: [luther-ttml-parsed.json](https://github.com/user-attachments/files/29399415/luther-ttml-parsed.json)

Also, if my TTML file is formatted with some XML parser, it would messed up the TTML parser and add unnecessary new line (`\n`) in the text

```json
{
    "start": 82889,
    "value": "It\ngo\nin,\nout,\ndo it\nreal\nslow\n\n(When you\nride it,\nslide)"
},
```

### Code of Conduct

- [x] I agree to follow Navidrome's Code of Conduct

### PR DESCRIPTION
**Pull request by deluan**
### Description

Fixes two TTML lyrics parsing defects that surface on Apple Music files which mix a main vocal with an `x-bg` (background) span group inside the same line — e.g. Kendrick Lamar's "luther", attached to the issue.

**1. Background cues corrupted the main cue's end time (`start == end`).**
Cue end-time normalization ran over the whole line's cue list in document order. Background cues are stored *after* the main cues but interleave *earlier* on the timeline, so the next-cue clamp pulled the last main cue's end down to its own start, producing `start == end` (e.g. the `slow` cue became `85934 → 85934` instead of `85934 → 86751`). End times are now normalized **per agent group**, the same way the Subsonic serializer already groups cues before building the response, so parallel vocal layers no longer clamp each other.

**2. Pretty-printed TTML injected spurious newlines.**
Whitespace between elements was treated as significant, so indented/reformatted TTML turned one line into many (`"It\ngo\nin,\nout,..."`). Per the TTML2 default `xml:space` handling (`linefeed-treatment="treat-as-space"`, `white-space-collapse="true"`), formatting whitespace now collapses to a single space, and hard line breaks come only from `<br/>` (tracked explicitly instead of via a raw `\n`).

The line-level `value` and per-agent `cueLine.value` deliberately remain the **full line text**, as required by the OpenSubsonic `songLyrics` v2 contract (where `cueLine.value` is "full text of the line, for validation/fallback"). The per-agent text is carried in each `cueLine`'s `cue[]` array — this part already matched the spec and is unchanged.

### Related Issues

Fixes #5671

### Type of Change

- [x] Bug fix
- [ ] New feature
- [ ] Documentation update
- [ ] Refactor
- [ ] Other (please describe):

### Checklist

- [x] My code follows the project's coding style
- [x] I have tested the changes locally
- [ ] I have added or updated documentation as needed
- [x] I have added tests that prove my fix/feature works (or explain why not)
- [x] All existing and new tests pass

### How to Test

1. Add a track with an Apple Music-style TTML sidecar containing a background (`x-bg`) span group, e.g. the [`03.luther.xml`](https://github.com/user-attachments/files/29399358/03.luther.xml) file from the issue (renamed to `.ttml`).
2. Scan the library, then call the OpenSubsonic endpoint with enhanced lyrics:
   `GET /rest/getLyricsBySongId.view?id=<songId>&enhanced=true&...&f=json`
3. Expected results for the `It go in, out, do it real slow (When you ride it, slide)` line:
   - The `slow` cue (main agent `v2`) reports `"end": 86751` — **not** `"end": 85934` (no `start == end`).
   - Both the main (`v2`) and background (`__nd_bg__|v2`) `cueLine` entries carry the full line text in `value`, with their own words in `cue[]`.
4. Edge case — feed the **pretty-printed** form of the same TTML (each `<span>` on its own indented line). The line `value` must be `"It go in, out, do it real slow (When you ride it, slide)"` with no embedded `\n`.
5. Edge case — a line using an explicit `<br/>` must still split into two lines.

### Additional Notes

- Two existing tests encoded the old newline-as-break behavior; they are corrected to expect collapsed whitespace. New tests cover whitespace collapse, `<br/>` preservation, and interleaved background cue timing.
- Minor incidental change: source files with two literal spaces between word spans (seen in the real file: `real slow  (When you`) now collapse to a single space in the line `value`, consistent with TTML `white-space-collapse` semantics.
- Verified end-to-end against a live instance through the real `getLyricsBySongId` v2 endpoint, including an A/B comparison against `master` (which reproduces both defects character-for-character) using the reporter's actual TTML file.

### CODE REVIEW

File: model/lyrics_ttml.go [hunk 1]
```diff
@@ -555,11 +552,33 @@ func finalizeTTMLLogicalLine(line []ttmlPiece) (string, []Cue) {
...
+	b.Grow(len(raw))
+	prevSpace := false
+	for _, r := range raw {
+		if unicode.IsSpace(r) {
+			if !prevSpace {
+				b.WriteByte(' ')
+				prevSpace = true
+			}
+			continue
+		}
```
[the remark this answers is not in the document]
Comment (deluan) [reply model/lyrics_ttml.go]: Fixed in 468618700 — collapsing now only touches the XML S characters (space/tab/CR/LF), so NBSP, U+3000 and other Unicode spaces pass through unchanged. Added a regression test covering U+3000.

File: model/lyrics_ttml.go [hunk 2]
```diff
@@ -555,11 +552,33 @@ func finalizeTTMLLogicalLine(line []ttmlPiece) (string, []Cue) {
...
+		b.WriteRune(r)
+		prevSpace = false
+	}
+	return b.String()
 }
```
[the remark this answers is not in the document]
Comment (deluan) [reply model/lyrics_ttml.go:587]: Same fix as above — collapsing is now restricted to the XML S set (space/tab/CR/LF) in 468618700, leaving NBSP and other Unicode spacing runes intact, with a regression test for U+3000.

File: model/lyrics_ttml.go [hunk 3]
```diff
@@ -555,11 +552,33 @@ func finalizeTTMLLogicalLine(line []ttmlPiece) (string, []Cue) {
...
+// and other whitespace runs collapse to a single space. Hard line breaks come
+// only from <br/> (§8.1.7), tracked separately via ttmlPiece.isBreak, so
+// pretty-printed indentation between elements does not inject spurious newlines.
 func normalizeTTMLPieceRaw(raw string) string {
 	raw = str.SanitizeText(raw)
-	raw = strings.ReplaceAll(raw, "\r\n", "\n")
-	raw = strings.ReplaceAll(raw, "\r", "\n")
-	return raw
+	return collapseTTMLWhitespace(raw)
+}
```
[the remark this answers is not in the document]
Comment (deluan) [reply model/lyrics_ttml.go]: The parser already collapsed whitespace unconditionally before this PR; nothing here introduces a new global policy. `xml:space="preserve"` has never been supported and no lyric source in practice relies on it, so I'm leaving that out (YAGNI) rather than threading a whitespace mode through. I did update the doc comment to state explicitly that collapsing is unconditional and `preserve` is unsupported (468618700).

File: model/lyrics_ttml.go [hunk 4]
```diff
@@ -382,7 +383,7 @@ func (p *ttmlParser) parseParagraph(parent ttmlTimingContext) (string, []Cue, er
 func (p *ttmlParser) parseInlineElement(start xml.StartElement, parent ttmlTimingContext) ([]ttmlPiece, error) {
 	local := strings.ToLower(start.Name.Local)
 	if local == "br" {
-		return []ttmlPiece{{raw: "\n"}}, nil
+		return []ttmlPiece{{isBreak: true}}, nil
```
[the remark this answers is not in the document]
Comment (deluan) [reply model/lyrics_ttml.go:386]: I checked this against master and it's not a regression: `<span begin="1s" end="3s">first<br/>second</span>` parses to `firstsecond` both before and after this change. A timed span flattens its children into a single cue via `concatTTMLPieceRaw`, and that path dropped the `<br/>` newline on master too. So the behavior is unchanged here. Worth fixing on its own, but it's pre-existing and out of scope for this PR.

File: model/lyrics_normalize.go [hunk 5]
```diff
@@ -85,10 +85,47 @@ func normalizeCueLine(line Line, fallbackEnd *int64) Line {
...
+	}
+
+	out := slices.Clone(cues)
+	for _, id := range order {
+		idxs := groups[id]
+		group := make([]Cue, len(idxs))
+		for gi, pos := range idxs {
+			group[gi] = cues[pos]
+		}
+		group = NormalizeCueEnds(group, fallbackEnd)
```
[the remark this answers is not in the document]
Comment (deluan) [reply model/lyrics_normalize.go:121]: Verified against master: the start-only sequential case you describe already produces the same overlap before this PR (v1 'A' ends at the line end, v2 'B' starts earlier). Per-agent normalization doesn't change it — single-agent normalization on master never clamped 'A' to 'B's start across different agents either. So this isn't introduced here. Cross-voice end inference for sequential duets is a separate question and out of scope for this fix.
